src/bun.js/api/bun/socket.zig

```diff
@@ -1835,6 +1872,7 @@ fn NewSocket(comptime ssl: bool) type {
             const callback = handlers.onEnd;
             if (callback == .zero or handlers.vm.isShuttingDown()) {
                 this.poll_ref.unref(handlers.vm);
+                this.poll_ref_before_connect = 0;
```

Comment:
Why is the poll ref in use at all before the socket is attached?

Reply:
likely to be extra sure about deinitialization, fixing